File: lib/src/main/java/com/darklycoder/wifitool/lib/info/action/WiFiNormalConnectAction.java
package com.darklycoder.wifitool.lib.info.action;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.darklycoder.wifitool.lib.type.WiFiCipherType;
import com.darklycoder.wifitool.lib.interfaces.ConnectWiFiActionListener;

/**
 * 通过密码连接WiFi
 */
public class WiFiNormalConnectAction extends WiFiConnectAction {

    public WiFiCipherType cipherType;
    public String password;

    public WiFiNormalConnectAction(@NonNull String SSID, @NonNull WiFiCipherType cipherType, @Nullable String password, @Nullable ConnectWiFiActionListener listener) {
        super(SSID, listener);

        this.cipherType = cipherType;
        this.password = password;
    }

    @NonNull
    @Override
    public String toString() {
        return super.toString() + " | " + SSID + " | " + cipherType.name();
    }
}

File: lib/src/main/java/com/darklycoder/wifitool/lib/info/action/WiFiNormalConnectAction.kt
package com.darklycoder.wifitool.lib.info.action

import com.darklycoder.wifitool.lib.interfaces.ConnectWiFiActionListener
import com.darklycoder.wifitool.lib.type.WiFiCipherType

/**
 * 通过密码连接WiFi
 */
class WiFiNormalConnectAction(
        SSID: String,
        var cipherType: WiFiCipherType,
        var password: String? = null,
        listener: ConnectWiFiActionListener? = null

) : WiFiConnectAction(SSID, listener) {

    override fun toString(): String {
        return "${super.toString()} | $SSID | ${cipherType.name} | $password"
    }

}
